src/components/image/image.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import {objectFitSupported} from '../../common/environment';
+import {observable, action} from 'mobx';
+import {observer} from 'mobx-react';
+import {SyntheticEvent} from "react";
+
+// Transparent 1x1 gif image
+export const onePixelTransparentSrc: string = 'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7';
+
+export interface ImageProps extends React.HTMLAttributes<HTMLImageElement> {
+    defaultImage: string;
+    title: string;
+    src: string;
+    alt: string;
+
+    onLoad: (data: object) => void;
+    onError: (data: object) => void;
+}
+
+@observer
+export class Image extends React.Component<Partial<ImageProps>, {}>{
+    static defaultProps: Partial<ImageProps> = {
+        defaultImage: onePixelTransparentSrc,
+        title: ''
+    };
+
+    @observable src: string = this.props.src ? this.props.src : this.props.defaultImage!;
+
+    @action setSrcToFallback = () => {
+        this.src = this.getFallbackSrcFor(this.src)!;
+    };
+
+    onError: React.EventHandler<SyntheticEvent<HTMLImageElement>> = () => {
+        if (this.props.onError) {
+            this.props.onError({data: this.src});
```

Comment:
The best practice is to use instance of error as argument of `onError`'s like functions.

You can create own error class if you need to provide more than stack and message
```ts
class ImageError extends Error {
    constructor(src) { 
        super('Image Error')
        this.src = src
    }
    src: string
 }
```
and inside `onError` you can call `onError(new ImageError(this.src))`